hammers/scripts/lease_stack_reaper.py

```diff
@@ -41,21 +42,19 @@ def __init__(self, user_id, name, email):
         self.user_id = user_id
         self.name = name
         self.email = email
-        self.nodes = {}
+        self.nodes = collections.defaultdict(list)
 
-    def add_lease(self, node_type, lease_id, start_date, end_date, **kwargs):
+    def add_lease(self, node_type, lease):
         """Add lease to node list."""
-        if node_type not in self.nodes:
-            self.nodes[node_type] = set()
-
-        self.nodes[node_type].add((lease_id, start_date, end_date))
+        self.nodes[node_type].append(lease)
 
     def sort_leases_by_date(self):
         """Sort leases by date for each node."""
         for node_type, leases in self.nodes.items():
-            self.nodes[node_type] = list(sorted(leases, key=lambda x: x[1]))
+            self.nodes[node_type] = list(
+                sorted(leases, key=lambda x: x['start_date']))
```

Comment:
Looks like this is going to sort the leases by their start_date as a string, probably it should be a dateobj. I think due to Blazar's date formatting technically the string sort would keep the right order, but it's a dangerous assumption over time.